### PR TITLE
[378980] Make line delimiter for Manifest configurable by CodeConfig

### PR DESCRIPTION
**Pull request by kthoms**
Line delimiter is determined by CodeConfig and reached into MergeableManifest.

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=378980

**Comment by meysholdt** [comment]
\r\n is hard-coded in java.util.jar.Manifest.write(OutputStream), even though the [spec](http://docs.oracle.com/javase/7/docs/technotes/guides/jar/jar.html) allows for \r\n and \n

**Comment by kthoms** [comment]
Right, I was not aware of that. So does this mean since standard Java Manifest uses \r\n it is also mandatory for MergeableManifest to do the same? Seems so, i.e. I should close the bug as WONTFIX?

**Comment by meysholdt** [comment]
Further investigation shows that `java.util.jar.Manifest.read(InputStream)` supports both `\r\n` and `\n`. Also, the Eclipse Manifest Editor supports both `\r\n` and `\n\`. Furthermore, the `MANIFEST.MF`s in xtext-core only contain `\n`. So I'm fine with making line-edings configurable in `MergeableManifest`. 

Can you confirm that your change doesn't mix line-edings, e.g. because a method from the super-class (`Manifest`) adds `\r\n` but in `MergeableManifest` `\n` is configured?

**Comment by kthoms** [comment]
screenshot 5

This is how it looks like after generating. The example above is without any configuration, it will use Strings.newLine() by default which is "\n". You can see all line endings are such.

The second example shows explicit configuration of "\r\n". Again, all line endings are the same.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java [hunk 1]
```diff
@@ -50,38 +50,47 @@
 	public static final Attributes.Name BUNDLE_LOCALIZATION = new Attributes.Name("Bundle-Localization");
 	public static final Attributes.Name BUNDLE_ACTIVATOR = new Attributes.Name("Bundle-Activator");
 
-	private static final String LINEBREAK = "\r\n";
+	private static final String LINEBREAK = Strings.newLine();
```
Comment (meysholdt) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: this filed is only used by deprecated methods, right? Then it should be deprecated itself.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/ManifestAccess.xtend [hunk 2]
```diff
@@ -38,6 +38,8 @@ class ManifestAccess extends TextFileAccess implements IGuiceAwareGeneratorCompo
 	
 	TypeReference activator
 	
+	String lineDelimiter = "\r\n"
```
Comment (meysholdt) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/ManifestAccess.xtend]: I think here the configuration from `CodeConfig` should be considered.

Comment (kthoms) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/ManifestAccess.xtend]: Not really. It is just the default value, which is overwritten by setLineDelimiter (available through @Accessors). But the default should be Strings.newLine(). I'll change that.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java (hunk 1 above)
Comment (kthoms) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest.java]: Correct, it is only used in deprecated contexts. Will fix that.

File: org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/ManifestAccess.xtend (hunk 2 above)
Comment (meysholdt) [inline org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/model/ManifestAccess.xtend]: ok

**Comment by meysholdt** [comment]
thx!

**Comment by kthoms** [comment]
Thanks for the review :)

**Pull request merged**